### PR TITLE
feat(k8s): add uninstall-garden-services command

### PR DESCRIPTION
**Pull request by edvald**
This cleans up everything in the garden-system namespace, as well as
related cluster-wide resources.

### CODE REVIEW

File: docs/using-garden/remote-kubernetes.md [hunk 1]
```diff
@@ -55,6 +55,14 @@ To initialize or update your cluster-wide services, run:
 garden --env=<environment-name> plugins kubernetes cluster-init
 ```
 
+To later uninstall the installed services, you can run:
+
+```sh
+garden --env=<environment-name> plugins kubernetes uninstall-cluster-services
```
Comment (eysi09) [inline docs/using-garden/remote-kubernetes.md]: Should be `uninstall-garden-services`

**Comment by eysi09** [comment]
Also, didn't delete the metadata namespace but I guess that's on the `cleanupEnvironment` handler.

**Comment by edvald** [comment]
Yeah I wasn't sure if we wanted to delete the metadata namespace, so I skipped that. That's an easy additional step if you want to clear out test results etc.

**Comment by edvald** [comment]
And yes, perhaps we should do that in the `cleanupEnvironment` handler

**Comment by edvald** [comment]
Should we make the timeout longer when deleting services? Or just leave it as is?

File: docs/using-garden/remote-kubernetes.md (hunk 1 above)
Comment (edvald) [reply docs/using-garden/remote-kubernetes.md]: Fixed

**Comment by eysi09** [comment]
I guess it's fine to increase it if it's likely to take a bit longer. You don't necessarily expect a delete action to fail fast.

**Comment by edvald** [comment]
Actually, the thing timing out was likely `helm delete ...` in this case. Could you paste the actual log output?

**Comment by eysi09** [comment]
Here you go:
```

[2019-07-18T14:49:25.632Z] RequestError: Error: read ETIMEDOUT
    at new RequestError (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request/request.js:185:22)
    at Request.emit (events.js:182:13)
    at Request.onRequestError (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
Error Details:
name: RequestError
cause:
  errno: ETIMEDOUT
  code: ETIMEDOUT
  syscall: read
error: ~cause
options:
  url: >-
    https://proxy.containership.io/v3/organizations/e7ea27a8-192d-4e08-a8f4-cf54c2ea5194/clusters/45d4bb8c-9c8f-4c25-8578-5280db964d98/k8sapi/proxy/api/v1/namespaces/garden-system/configmaps/garden-docker-registry-config
  method: GET
  json: true
  resolveWithFullResponse: true
  headers: []
  simple: true
  transform2xxOnly: false
```

**Comment by eysi09** [comment]
Hmm, it just timed out `get`-ing the ConfigMap:
```
[2019-07-18T14:49:25.632Z] RequestError: Error: read ETIMEDOUT
    at new RequestError (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request-promise-core/lib/errors.js:14:15)
    at Request.plumbing.callback (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request-promise-core/lib/plumbing.js:87:29)
    at Request.RP$callback [as _callback] (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request-promise-core/lib/plumbing.js:46:31)
    at self.callback (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request/request.js:185:22)
    at Request.emit (events.js:182:13)
    at Request.onRequestError (/Users/eysi/code/garden-io/garden/garden-service/node_modules/request/request.js:881:8)
    at ClientRequest.emit (events.js:182:13)
    at TLSSocket.socketErrorListener (_http_client.js:392:9)
    at TLSSocket.emit (events.js:182:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
Error Details:
name: RequestError
cause:
  errno: ETIMEDOUT
  code: ETIMEDOUT
  syscall: read
error: ~cause
options:
  url: >-
    https://proxy.containership.io/v3/organizations/e7ea27a8-192d-4e08-a8f4-cf54c2ea5194/clusters/45d4bb8c-9c8f-4c25-8578-5280db964d98/k8sapi/proxy/api/v1/namespaces/garden-system/configmaps/garden-docker-registry-config
  method: GET
  json: true
  resolveWithFullResponse: true
  headers: []
  simple: true
  transform2xxOnly: false
```

**Comment by edvald** [comment]
That looks like a provider specific thing. Are you sure that's the right message?

**Comment by eysi09** [comment]
Yeah that's the one. Looking at my terminal again, it doesn't appear to be related to the `uninstall` command: 
```
➜  demo-project (add-uninstall-command.) garden --env=remote plugins kubernetes uninstall-garden-services
Removing cluster-wide services for remote environment
    Deleting services...
      ✔ default-backend           → Not found
       docker-registry           → Deleting...
      ✔ registry-proxy            → Service deleted
      ✔ nfs-provisioner           → Service deleted
      ✔ kubernetes-dashboard      → Not found
      ✔ build-sync                → Service deleted
      ✔ docker-daemon             → Not found
      ✔ ingress-controller        → Not found
Error: read ETIMEDOUT

See error.log for detailed error message
```

(We need to fix the log output for the plugin commands btw)

**Comment by eysi09** [comment]
So I'll go ahead and approve this one.

**Pull request merged**